core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3570,6 +3591,72 @@ public long generateSeqNo() {
         }
     }
 
+    public void testMinGenerationForSeqNo() throws IOException, BrokenBarrierException, InterruptedException {
```

Comment:
I also pushed f5d3973dd7714ac98717c4a376a11c81d1e74b49 to add more assertions around the commit metadata. Regarding the recovery, I do not see what can be added that is not already covered by other tests. Is there something that you have in mind?